packages/DataGrid/src/DataGrid.js

```diff
@@ -0,0 +1,622 @@
+  }, []);
+
+  const handleScroll = React.useCallback(parameters => {
+    const { scrollLeft, scrollTop } = parameters;
+
+    if (refScrollHeader.current) {
+      refScrollHeader.current.scrollTo({ left: scrollLeft, top: 0 });
+    }
+
+    // prevent rescrolling when this scrollbar gets sync with the one in the sticky column
+    if (refScrollHappenedBy.current === null) {
+      refScrollHappenedBy.current = "handleScroll";
+      if (refScrollStickyColumns.current) {
+        refScrollStickyColumns.current.scrollTo({ left: 0, top: scrollTop });
+      }
+    } else {
+      refScrollHappenedBy.current = null;
+    }
+
+    if (
+      refScrollGrid.current &&
+      refEnd.current &&
+      refScrollGrid.current.scrollTop + refScrollGrid.current.offsetHeight >= refScrollGrid.current.scrollHeight
+    ) {
+      refEnd.current.onScrollBarReachedBottom(true);
+    } else if (refEnd.current) {
+      refEnd.current.onScrollBarReachedBottom(false);
+    }
+  }, []);
+
+  const handleScrollStickyColumns = React.useCallback(
+    parameters => {
+      const { scrollTop /* scrollUpdateWasRequested */ } = parameters;
+
+      // prevent rescrolling when this scrollbar gets sync with the one in the main grid
+      if (refScrollHappenedBy.current === null) {
+        refScrollHappenedBy.current = "handleScrollStickyColumns";
+        if (refScrollGrid.current) {
+          refScrollGrid.current.scrollTo({ left: 0, top: scrollTop });
+        }
+      } else {
+        refScrollHappenedBy.current = null;
+      }
+    },
+    [refScrollGrid]
+  );
+
+  const handleRefCell = React.useCallback(({ columnIndex, rowIndex }) => {
+    return ref => {
+      const key = `${columnIndex}${rowIndex}`;
+      refsCell.current.keys[key] = ref;
+      refsCell.current.rows[rowIndex] = Array.isArray(refsCell.current.rows[rowIndex])
+        ? refsCell.current.rows[rowIndex].concat(key)
+        : [key];
+    };
+  }, []);
+
+  React.useEffect(() => {
+    if (!refContainer.current) return;
+
+    refScrollHeader.current = refContainer.current.querySelector(`.${gridId}-header`);
+    refScrollStickyColumns.current = refContainer.current.querySelector(`.${gridId}-sticky-columns`);
+    refScrollGrid.current = refContainer.current.querySelector(`.grid-${gridId}`);
+  }, [gridId]);
+
+  React.useEffect(() => {
+    const scrollContainer =
+      refContainer.current && refContainer.current.querySelector(`.grid-${gridId} [role="row"]`).parentElement;
+
+    if (!scrollContainer) return;
+    // https://stackoverflow.com/questions/21064101/understanding-offsetwidth-clientwidth-scrollwidth-and-height-respectively
+    // https://davidwalsh.name/detect-scrollbar-width
+    setScrollBarWidth(() => scrollContainer.offsetWidth - scrollContainer.clientWidth);
+  }, [gridId, isIdle]);
+
+  function handleFocusGrid() {
+    const $isBlurred = refContainer.current.querySelector(".grid--is-blurred");
```

Comment:
well is not really arbitrary, is returning a dom element which tends to be prefixed with a `$` to communicate their meaning. 
I guess on react that is not too common since normally you shouldn't need to deal with bare Dom elements.